src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationProvider.cs

```diff
@@ -587,10 +634,17 @@ private async Task InitializeAsync(bool ignoreFailures, IEnumerable<Configuratio
 
                 try
                 {
-                    Dictionary<string, ConfigurationSetting> mappedData = await MapConfigurationSettings(data).ConfigureAwait(false);
-                    SetData(await PrepareData(mappedData, cancellationToken).ConfigureAwait(false));
+                    Dictionary<KeyValueIdentifier, ConfigurationSetting> mappedData = await MapConfigurationSettings(data).ConfigureAwait(false);
```

Comment:
I dont think we need `mappedData` here. We can directly assign to `_mappedData`.